teamscale_client/client.py

```diff
@@ -108,157 +142,205 @@ def put(self, url, json=None, parameters=None, data=None):
         """Adds descriptions of findings.
 
         Args:
-            descriptions (list): List of :class:`FindingDescription` to add to Teamscale.
+            descriptions: List of :class:`FindingDescription` to add to Teamscale.
         Returns:
             requests.Response: request's response
         """
-        base_url = self.get_global_service_url('external-findings-description')
         response = None
         for finding_description in descriptions:
-            some_description = dict()
-            some_description['typeId'] = finding_description.typeid
-            some_description['description'] = finding_description.description
-            some_description['enablement'] = finding_description.enablement
-            some_description['name'] = finding_description.name
-            url = "%s/%s" % (base_url, finding_description.typeid)
-            response = self.put(url, some_description)
-            if response.text != 'success':
-                return response
-
+            response = self.post(
+                f"{self._api_url_version}/external-findings/descriptions",
+                data=to_json(finding_description),
+                headers={'Accept': 'application/json', 'Content-Type': 'application/json'}
+            )
         return response
 
     def update_findings_schema(self):
-        """Triggers refresh of finding groups in analysis profiles."""
-        url = self.get_global_service_url('update-findings-schema')
-        return self.get(url, {'projects': self.project})
+        """Triggers refresh of finding groups in analysis profiles.
 
-    def upload_findings(self, findings, timestamp, message, partition):
+        Warnings:
+            Uses Teamscale's internal API and not the versioned API!
+        """
+        return self.post(
+            f"{self._api_url}/projects/{self.project}/metric-update",
+            params={"projects": self.project},
+            headers={'Accept': 'application/json', 'Content-Type': 'application/json'}
+        )
+
+    def upload_findings(
+            self, findings: List[FileFindings], timestamp: datetime.datetime, message: str, partition: str
+    ) -> requests.Response:
         """Uploads a list of findings
 
         Args:
-            findings (List[:class:`data.FileFindings`]): the findings data 
-            timestamp (datetime.datetime): timestamp for which to upload the findings
-            message (str): The message to use for the generated upload commit
-            partition (str): The partition's id into which the findings should be added (See also: :ref:`FAQ - Partitions<faq-partition>`).
+            findings: the findings data
+            timestamp: timestamp for which to upload the findings
+            message: The message to use for the generated upload commit
+            partition: The partition's id into which the findings should be added
```

Comment:
We typically talk about partition names not IDs. I'd suggest to adjust the wording here and in other occurrences to avoid confusion